src/Sparrow/Json/JsonOperationContext.cs

```diff
@@ -403,47 +403,66 @@ private static void ThrowParentWasDisposed()
             }
         }
 
+
+#if DEBUG || VALIDATE
+        private readonly LongReference _getMemory = new LongReference();
```

Comment:
Why do we have multiple of them? We don't actually care if the concurrent usage is between methods or just in one method. I think we should have just one of these

Reply:
They call each other

Reply:
That being the case, let's do this differently and say that we want to do:
`Interlocked.CompareExchange(ref val, 0, CurrentThreadId)`

This way, in the same thread, it will work, but you can catch it if it is between threads.